### PR TITLE
no need delete endpoint explicitly in endpoint controller

### LINKED ISSUE #57745: endpoint controller no need to delete endpoint when service is deleted
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:

`EndpointController.syncService` call delete endpoint when service is not found. 

**What you expected to happen**:

It is no need to do delete in controller. Because in service restStorage, endpoint is deleted when delete service. See https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/rest.go#L170~L180

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

It is no need to do delete in controller. Because in service restStorage, endpoint is deleted when delete service. See https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/rest.go#L170~L180

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #57745

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/endpoint/endpoints_controller.go [hunk 1]
```diff
@@ -394,15 +394,7 @@ func (e *EndpointController) syncService(key string) error {
...
-		// Delete the corresponding endpoint, as the service has been deleted.
-		// TODO: Please note that this will delete an endpoint when a
-		// service is deleted. However, if we're down at the time when
-		// the service is deleted, we will miss that deletion, so this
-		// doesn't completely solve the problem. See #6877.
-		err = e.client.CoreV1().Endpoints(namespace).Delete(name, nil)
-		if err != nil && !errors.IsNotFound(err) {
-			return err
-		}
+		// Service has been deleted. So no need to do any more operations.
```
Comment (deads2k) [inline pkg/controller/endpoint/endpoints_controller.go]: Since we always queue based on the service lister (even when adding pods), I think this works.

**Comment by MrHohn** [comment]
Would the remain TODO (which leaves dangling endpoints) be a concern?
https://github.com/kubernetes/kubernetes/blob/91615e4fd9150c1e75dde331d3d00be0e26bff69/pkg/registry/core/service/rest.go#L170-L180

This also make `checkLeftoverEndpoints()` ineffective, or maybe we don't need that anymore? https://github.com/kubernetes/kubernetes/blob/ad0674702713fc36ebe53c99f2bab1d54d7a82ff/pkg/controller/endpoint/endpoints_controller.go#L525-L554

**Comment by hzxuzhonghu** [comment]
@MrHohn I missed the `checkLeftoverEndpoints`, strictly speaking, the function now does not work.

**Comment by MrHohn** [comment]
@freehan @thockin Are we ok with standalone endpoints not being cleaned up?

**Comment by hzxuzhonghu** [comment]
@MrHohn I am also worried about if service is deleted but it fails when delete endpoint in apiserver.

**Comment by zhangxiaoyu-zidif** [comment]
@MrHohn @freehan @thockin I agree that endpoint could be  standalone. In kubernetes, personally speaking, every resource objects should not have too tight relationships with each other. If that many more possible usage situations would be created. that's my wish. thanks =)

**Pull request merged**